Bio/SeqIO/pepdigestIO.py

```diff
@@ -0,0 +1,143 @@
+# package.
+#
+# This module is for reading .pepdigest format files as SeqRecord objects.
+"""Bio.SeqIO support for the "pepdigest" file format.
+
+You are expected to use this module via the Bio.SeqIO functions.
+"""
+import re
+
+from Bio.Seq import Seq
+from Bio.SeqRecord import SeqRecord
+
+from .Interfaces import SequenceIterator
+
+
+class PepdigestIterator(SequenceIterator):
+    """TODO: class docstring."""
+
+    main_header_pttrn = re.compile("#{2,}")
+    digestion_title_pttrn = re.compile("#=+")
+    max_iter = 100
+
+    def __init__(self, source):
+        """Break up a pepdigest seqtable file into SeqRecord objects.
+
+        Argument source is a file-like object opened in text mode or a path to a file.
+        """
+        return super().__init__(source)
+
+    def parse(self, handle):
+        """Start parsing the file, and return a SeqRecord generator."""
+        # Reads the main header
+        header_sep_count = 0
+        i = 0
+        while header_sep_count < 2:
+            line = handle.readline()
+            if self.main_header_pttrn.search(line) is not None:
+                header_sep_count += 1
+            i += 1
+            if i > self.max_iter:
+                raise ValueError("Broken header in start of file.")
+
+        records = self.iterate(handle)
+
+        return records
+
+    def parse_digestion_header(self, handle):
+        """Parse the header of a digestion.
+
+        Return usefull information for parsing the results.
+        """
+        # Read sequence name and positions
+        i = 0
+        line = handle.readline()
+        while not line.startswith("# Sequence"):
+            line = handle.readline()
+            i += 1
+            if i > self.max_iter:
+                raise ValueError(
+                    "Can't find '# Sequence' section in digestion results header"
+                )
+
+        line = line.rsplit("# Sequence:")[1].strip()
+        orig_seq_name, line = line.split(None, 1)
+        line, seq_to = line.split("to:")
+        seq_to = int(seq_to)
+        seq_from = int(line.replace("from:", ""))
+
+        # Read HitCount
+        line = handle.readline()
+        hitcount = int(line.replace("# HitCount:", ""))
+
+        # Read enzyme utilized in digestion
+        handle.readline()
+        line = handle.readline()
+        enzyme = line.split("with")[1].split("yields")[0].strip()
+
+        # Read end of header
+        i = 0
+        while self.digestion_title_pttrn.search(line) is None:
+            line = handle.readline()
+            i += 1
+            if i > self.max_iter:
+                raise ValueError("Malformed digestion results header")
+
+        return orig_seq_name, seq_to, seq_from, hitcount, enzyme
+
+    def parse_digestion_results(self, handle, header):
+        """Parse the digested peptides."""
+        orig_seq_name, seq_to, seq_from, hitcount, enzyme = header
+        # skip one line
+        handle.readline()
```

Comment:
Should this line be blank? Why not check it?

Reply:
Hm, and raise a `ValueError` if it contains something?